keps/sig-cli/5091-kubectl-checkpoint/README.md

```diff
@@ -0,0 +1,1205 @@
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+Introduce the `kubectl checkpoint` sub-command to allow users to checkpoint a
+container. The *kubelet* already provides the possibility to checkpoint a
+container and this *kubelet* only API endpoint is extended to *kubectl* for
+easier user consumption.
+
+## Motivation
+
+The introduction of KEP-2008 (Forensic Container Checkpointing) in Kubernetes
+1.25 was a *kubelet* only change initially as checkpointing containers is a
+completely new concept in the context of Kubernetes. From the beginning it
+was slightly confusing for users that it was only a *kubelet* API endpoint.
+Now with KEP-2008 (Forensic Container Checkpointing) graduating from Alpha to
+Beta in Kubernetes 1.30, which means that the corresponding feature gate
+defaults to the feature being enabled, the next step would be to extend the
+existing checkpointing functionality from the *kubelet* to *kubectl* for easier
+user consumption. The main motivation is to make it easier by not requiring
```

Comment:
I'm not familiar with the criteria followed for kubectl commands, but should not wait first for the feature to be GA so it is available? or is the plan to GA KEP-2008 and add the kubectl command at the same time?

Reply:
I also don't know. Not sure. But as there is a possibility to add plugins for `kubectl` or `alpha` commands I thought it is possible to expose non GA features on the api server level. But I don't know.